### DISCUSSION TITLE
(loop [x :range [1 nil 2]]) now broken, used to count odd numbers forever

### DISCUSSION THREAD

**Discussion by jackpipe** [janet-lang/janet#1324] [General]
As the title, I have a prime-number generator using a fiber/generator/coroutine and this was a useful trick to not have to specify an upper limit on the main loop.

Now it runs the loop once with a value of 0 and then quits? Not sure, but it doesn't do what it used to.

**Comment by sogaiu** [comment]
AFAICT, it works in [this commit](https://github.com/janet-lang/janet/commit/874fd2aba7af3d8402704c4052a964a7767db76b), but not in [the one after](https://github.com/janet-lang/janet/commit/2f43cb843e15e7a1fccb042886a33f017b805846).

---

FWIW, on a related note, there is some prime-number-related functionality in spork now:

* [Add functions for primality testing](https://github.com/janet-lang/spork/pull/145)
* [Add math/factor](https://github.com/janet-lang/spork/pull/152)

**Comment by primo-ppcg** [comment]
Quick fix:
```janet
repl:1:> (def g (generate [x :range [1 () 2]] x))
<fiber 0x5620123A2760>
repl:2:> (take 10 g)
@[1 3 5 7 9 11 13 15 17 19]
```
Works because:
```janet
repl:3:> (< 1 nil)
true
repl:4:> (< 1 ())
true
```
I don't know that I'd recommend doing this, though. Using `forever` with an increment would avoid an unnecessary comparison each iteration, and would be more clear.
